e2e/server_test.go

```diff
@@ -185,6 +187,7 @@ func TestProxyServerWithOversizedBlob(t *testing.T) {
 	}
 
 	require.True(t, oversizedError)
+	require.Contains(t, err.Error(), fmt.Sprint(http.StatusBadRequest))
```

Comment:
Can't you use `require.ErrorIs(t, err, http.StatusBadRequest)` here?
https://pkg.go.dev/github.com/stretchr/testify/require#ErrorIs

Reply:
don't think we can since it assumes the string contents to be identical

Reply:
It says that it works the same way as `errors.Is`, so this should work now that we're wrapping errors I would think? But either works honestly